[RiyaMehta2211] iP

RiyaMehta2211 opened the pull request:
### **Harry Potter**
> “Yer a planner, Harry” – Hagrid in the alternate universe 2.0

Harry Potter promises to provide the perfect, wizarding experience, your trusted companion and guide to efficient and effective magical planning!
- User-friendly
- Reliable in saving your task lists!
- _TRULY **EFFICIENT** & **EFFECTIVE**_ 
- AND MAGICAL ✨✨

How to get started?
1. Click [here](https://nus-cs2103-ay2324s1.github.io/website/admin/ip-w4.html)
2. Run it
3. Add your tasks
4. Sit back and watch it manage your tasks for you!

What are you waiting for? 
⚡Get it ~~TODAY~~ **NOW!** ⚡

Features to look forward to:
- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Setting calendar reminders for events

If you are a Java programmer, we have just the solution for you! DukePro allows you to enhance your Java skills too by running the main method as shown here:
`public class Main {

    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }

}`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level-0

Commit:
Level-0

Commit:
Level-0

Commit:
Level-1

Commit:
Level-0

Commit:
Level-1

Commit:
Level-0

Commit:
Level-1

Commit:
commit message 0

Commit:
ip - Riya Mehta

Commit:
ip - Riya Mehta

Commit:
Committing Level 1

Commit:
Committing Level 3

Commit:
Committing Level 2

Commit:
Committing Level 2

Commit:
Committing Level 2

Commit:
Tagging Level 2

Commit:
Tagging Level 2

Commit:
Committing Level 3

Commit:
Committing Level 3

Commit:
Committing Level 3

Commit:
Committing Level 3

Commit:
Committing Level 3

Commit:
Committing Level 3

Commit:
Committing Level 3

Commit:
Committing Level 3

Commit:
Committing Level 4

Commit:
Committing Level 4

Commit:
Committing text-ui-test

Commit:
Committing Level 4

Commit:
Committing A-TextUiTesting

Commit:
Committing Level-5

Commit:
Committing Level-6

Commit:
Save the tasks in the hard disk automatically whenever the task list changes for Level 7. The tasklist will be saved to Duke.txt in UTF-16 encoded format when terminating the chatBot. The saved taskList can be retrieved after restarting the chatBot.

Commit:
Save the tasks in the hard disk automatically whenever the task list changes for Level 7. The tasklist will be saved to Duke.txt in UTF-16 encoded format when terminating the chatBot. The saved taskList can be retrieved after restarting the chatBot.

Commit:
Save the tasks in the hard disk automatically whenever the task list changes for Level 7. The tasklist will be saved to Duke.txt in UTF-8 encoded format when terminating the chatBot. The saved taskList can be retrieved after restarting the chatBot.

Commit:
Save the tasks in the hard disk automatically whenever the task list changes for Level 7. The tasklist will be saved to Duke.txt in UTF-8 encoded format when terminating the chatBot. The saved taskList can be retrieved after restarting the chatBot.

Commit:
Save the tasks in the hard disk automatically whenever the task list changes for Level 7. The tasklist will be saved to Duke.txt in UTF-8 encoded format when terminating the chatBot. The saved taskList can be retrieved after restarting the chatBot.

Commit:
Save the tasks in the hard disk automatically whenever the task list changes for Level 7. The tasklist will be saved to Duke.txt in UTF-8 encoded format when terminating the chatBot. The saved taskList can be retrieved after restarting the chatBot.

Commit:
Save the tasks in the hard disk automatically whenever the task list changes for Level 7. The tasklist will be saved to Duke.txt in UTF-8 encoded format when terminating the chatBot. The saved taskList can be retrieved after restarting the chatBot.

Commit:
Save the tasks in the hard disk automatically whenever the task list changes for Level 7. The tasklist will be saved to Duke.txt in UTF-8 encoded format when terminating the chatBot. The saved taskList can be retrieved after restarting the chatBot.

Commit:
Save the tasks in the hard disk automatically whenever the task list changes for Level 7. The tasklist will be saved to Duke.txt in UTF-8 encoded format when terminating the chatBot. The saved taskList can be retrieved after restarting the chatBot.

Commit:
Save the tasks in the hard disk automatically whenever the task list changes for Level 7. The tasklist will be saved to Duke.txt in UTF-8 encoded format when terminating the chatBot. The saved taskList can be retrieved after restarting the chatBot.

Commit:
The deadline and event tasks now require a concrete dd-MM-yyyy HHmm format input, and will output in MMM d yyyy HHmm format (time in the 24 hours format) as part of Level 8.

Commit:
The deadline and event tasks now require a concrete dd-MM-yyyy HHmm format input, and will output in MMM d yyyy HHmm format (time in the 24 hours format) as part of Level 8.

Commit:
Revert "The deadline and event tasks now require a concrete dd-MM-yyyy HHmm format input, and will output in MMM d yyyy HHmm format (time in the 24 hours format) as part of Level 8."

This reverts commit 0d460fbc974a67092335a021ee6dd898f1ad013e.

Commit:
Revert "Revert "The deadline and event tasks now require a concrete dd-MM-yyyy HHmm format input, and will output in MMM d yyyy HHmm format (time in the 24 hours format) as part of Level 8.""

This reverts commit 2861c63a7dcff160e015c3b920851ef0522f1abb.

Commit:
Import LocalDateTime to parse user inputs for date and time of tasks

Commit:
Merge branch-Level-8 with master

Commit:
Add the header comment of the Task Class

Commit:
merge branch-Level-7 with master

Commit:
merge branch-Level-8 with master

Commit:
Fix minor string formatting issues in the Deadline Class

Commit:
create Ui class for the chatBot which interacts with users

Commit:
create Storage class for the chatBot to load and save tasks in the file

Commit:
create Parser class for the chatBot to accept and read user inputs

Commit:
create Parser class for the chatBot to parse and read user inputs

Commit:
create Ui class for the chatBot to interact with user inputs

Commit:
create TaskList class for the chatBot to handle the tasklists created

Commit:
create packages - duke, task and exception to categorize the classes in suitable packages

B-enguin commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,27 @@
+
+//This class has a chatbot which will generate a list based on a set of tasks
+//that have been input by the user. These tasks include deadlines, events and ToDos
+//The user input will be parsed to extract the date and time of the events and deadlines
```
Perhaps a multiline comment might be better, e.g.
```suggestion
/**
 * This class has a chatbot which will generate a list based on a set of tasks
 * that have been input by the user. These tasks include deadlines, events and ToDos
 * The user input will be parsed to extract the date and time of the events and deadlines
 */
```

B-enguin commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+        Ui ui = new Ui();
+        try {
+                if (!str.equals("list")) {
```
Perhaps the try block should have an indentation level of 4 rather than 8

B-enguin commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+        } catch (java.time.format.DateTimeParseException e) {
+            //detect inputs that don't follow the yyyy-MM-dd HHmm format
+            ui.printException();
+        } catch (InvalidInputException | EventCommandUseException |
+                 DeadlineCommandUseException | ToDoCommandUseException e) {
+            ui.printException(e.getMessage());
```
Perhaps you should break the line before the operator symbol, e.g.
```suggestion
        } catch (InvalidInputException | EventCommandUseException
                | DeadlineCommandUseException | ToDoCommandUseException e) {
            ui.printException(e.getMessage());
```

B-enguin commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,33 @@
+    public TaskList(){
+    }
+
+    public void addTask(Task task){
+        this.tasks.add(task);
+    }
+
+    public Task removeTask(int index){
+        return this.tasks.remove(index);
+    }
+    public Task getTask(int index){
```
Perhaps whether having a line-break between methods should be consistent

B-enguin commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    public void printDone(Task done) {
+        System.out.println("\t" + "Nice! I've marked this task " +
+                "as done:" + "\n" +
+                "\t " + done.taskString());
```
Perhaps break should be before the operator, like the methods below

RiyaMehta2211 replied:
Yup I see, thank you

B-enguin reviewed (commented):
Good job on the iP thus far, I like the use of OOP to split the different exceptions and tasks used in the application. Perhaps some Javadocs should be added to describe what each public methods does. Do take note of line breaks to enhance the readability of the code.

Commit:
merge add-gradle-support into master

0xteet0 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,34 @@
+package duke.task;
+
+//The duke.task.Task class is in charge of creating task objects
```
Good attempt on the JavaDocs! 👍  However, it might be great to notice that JavaDoc comments should have the following form as per the JavaDocs standards:

```Java
/**
 * Returns lateral location of the specified position.
 * If the position is unset, NaN is returned.
 *
 * @param x X coordinate of position.
 * @param y Y coordinate of position.
 * @param zone Zone of position.
 * @return Lateral location.
 * @throws IllegalArgumentException  If zone is <= 0.
 */
```

0xteet0 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+        //empty constructor to initialize class objects
+    }
+
+    public void chat(String str, TaskList tasks) {
```
Perhaps the chat method could be simplified, as to avoid it being a long method, which makes reading the code more difficult 😊

RiyaMehta2211 replied:
yup, I see what you mean. Thank you.

0xteet0 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+                                String byWhen = "/by ";
+                                int index = str.indexOf(byWhen);
+                                String deadline = str.substring(index + 4); //remove /by from the substring
+                                if (deadline.trim().isEmpty()) { //needs to check whether there is anything after /by
```
Is there any considerations as to the deep nesting of logic here? Perhaps consider avoid deep nesting within this method as it makes keeping track of logic difficult! 😊

0xteet0 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,126 @@
+                    } else {
+                        if (str.startsWith("todo")) {
+                            String todo = str.substring(4);
+                            //remove any leading and trailing whitespace characters and
```
Great on abiding the rule that comments should be indented relative to their position in the code! 😊

0xteet0 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,81 @@
+            String line;
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+            while ((line = reader.readLine()) != null) {
+                String[] components = line.split("\\|");
```
Good use of plural form on names representing a collection of object! 👍

RiyaMehta2211 replied:
Thank you!

0xteet0 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,81 @@
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.*;
```
Is there a use case for this wildcard import? Perhaps consider listing the imports explicitly as per the coding standards! 😊

0wl3xSec reviewed (commented):
Overall, great attempt! Perhaps consider fixing up the code as per coding standards and simplifying things to allow greater tracking of logic! Nonetheless, great attempt! 😊

Commit:
set up gradle and merge add-gradle-support with master

Commit:
Add JUnit tests for Storage, TaskList and Deadline classes

Commit:
Troubleshoot issues associated with JUnit and gradlew jar files

Commit:
Troubleshoot issues associated with JUnit and gradlew jar files

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Troubleshoot issues associated with JUnit and gradlew jar files

Commit:
Troubleshoot issues associated with JUnit and gradlew jar

Commit:
Troubleshoot issues associated with JUnit and gradle jar

Commit:
create a Duke.jar file with JUnit tests for Storage, TaskList and Deadline Classes

Commit:
Add JavaDocs to the Java classes

Commit:
Improve Coding Standards across the Java files

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Improve Coding Standards across the Java files

Commit:
Add Find instruction for chatBot to find matching results

Commit:
merge branch-Level-9 into master

RiyaMehta2211 committed:
Update src/main/java/duke/Parser.java

Co-authored-by: B-enguin <46633877+B-enguin@users.noreply.github.com>

Commit:
make changes to align with coding standards

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
troubleshoot and set up GUI

Commit:
set up GUI for Duke

Commit:
Merge remote-tracking branch 'origin/branch-Level-10' into branch-Level-10

# Conflicts:
#	build.gradle
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Launcher.java

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/duke/Parser.java

Commit:
merge branch-Level-10 into master after resolving conflicts

Commit:
new and improved GUI

Commit:
new modifications to GUI

Commit:
new modifications to GUI and merge branch-Level-10 into master

Commit:
Merge commit 'ebb695fee24deeb310bbef5b0eb30a2d243d003f' into branch-Level-10

* commit 'ebb695fee24deeb310bbef5b0eb30a2d243d003f':
  new and improved GUI
  merge branch-Level-10 into master after resolving conflicts
  Update src/main/java/duke/Parser.java

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:

Commit:
minor modifications to GUI and resolve conflicts between branches

Commit:
Merge remote-tracking branch 'origin/branch-Level-10' into branch-Level-10

Commit:
Add assertions for error handling and important assumptions for valid inputs

Exceptions and assertions are two complementary ways of handling exceptions and errors in software. Assert statements here help identify important string inputs such as filePaths which should not be null. Assert statements also help ensure integer inputs for indices are in the right range.

In order to identify more potential error-inducing code, let's extract out possible places for invalid inputs. And also use assertions and exception handling mechanisms appropriately to improve on the code design.

RiyaMehta2211 committed:
Merge pull request #1 from RiyaMehta2211/branch-A-Assertions

Add assertions for error handling and important assumptions for valid inputs

Commit:
Improve code quality by creating methods in Parser

The chat() method is in charge of parsing the different user inputs. The subsequent actions based on the respective inputs are split into different methods. The actions have been categorised under updating lists, adding tasks and returning lists. 3 methods to represent these categories.

Let's improve code quality by making more methods which are shorter and avoid deep nesting. It is also good to keep methods simple and easy to understand.

Commit:
Making minor modification to Event class taskstring

RiyaMehta2211 committed:
Merge pull request #2 from RiyaMehta2211/branch-A-CodeQuality

Improve code quality by making more readable methods

Commit:
Add C-help extension to provide in-App guidance to users

Commit:
Add C-MassOps extension to mark, unmark and delete all tasks in one go

Commit:
Add javadocs and align code towards the coding standards

Commit:
Add representative screenshot Ui.png in docs & a brief User Guide (UG) in README.md

RiyaMehta2211 committed:
Update README.md

Commit:
Update README.md

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
add C-update extension to update attributes of the specified task in the list. Add extra exceptions for more efficient error handling

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Update README.md

RiyaMehta2211 committed:
minor modification to README.md

Commit:
minor modification to README.md

Commit:
Merge remote-tracking branch 'origin/master'

Commit:
Minor modifications to GUI and align to coding standards

Commit:
Removing commented out bits to improve code quality

Commit:
Fix minor issues to align with coding standards and improve coding quality
